src/net_processing.cpp

```diff
@@ -2132,33 +2152,26 @@ void PeerManagerImpl::ProcessHeadersMessage(CNode& pfrom, const Peer& peer,
     return;
 }
 
-/**
- * Reconsider orphan transactions after a parent has been accepted to the mempool.
- *
- * @param[in,out]  orphan_work_set  The set of orphan transactions to reconsider. Generally only one
- *                                  orphan will be reconsidered on each call of this function. This set
- *                                  may be added to if accepting an orphan causes its children to be
- *                                  reconsidered.
- */
-void PeerManagerImpl::ProcessOrphanTx(std::set<uint256>& orphan_work_set)
+bool PeerManagerImpl::ProcessOrphanTx(Peer& peer)
 {
+    AssertLockHeld(m_mutex_message_handling);
```

Comment:
At commit f86a525, I don't think this lock is useful to protect `vExtraTxnForCompact`, there is already one in `AddToCompactExtraTransactions` ? Though it might be useful at branch tip ?

Reply:
I was just adding `AssertLockHeld` to correspond with the `EXCLUSIVE_LOCKS_REQUIRED` annotation as a matter of course (see the recommendation in developer-notes.md).